lucene/analysis/common/src/java/org/apache/lucene/analysis/miscellaneous/TypeAsSynonymFilter.java

```diff
@@ -64,9 +82,15 @@ public boolean incrementToken() throws IOException {
       }
       termAtt.append(typeAtt.type());
       posIncrAtt.setPositionIncrement(0);
+      // control what flags transfer to synonym
+      int flags = getAttribute(FlagsAttribute.class).getFlags();
```

Comment:
This should not call getAttribute and instead use the instance `flagsAtt` above (instance field). In the next line it's correct.